### ISSUE TITLE
Crash Upon Building - FileSystemException Deletion failed - build Directory

### ISSUE THREAD

**Issue by p30arena** [flutter/flutter#26158]
### I had to manually delete the build directory.

```
Flutter crash report; please file at https://github.com/flutter/flutter/issues.
```
## command
```
flutter build bundle --suppress-analytics --target M:\lab\gilda\yabco_divar\lib\main.dart --filesystem-scheme org-dartlang-root --track-widget-creation --compilation-trace-file compilation.txt --depfile M:\lab\gilda\yabco_divar\build\app\intermediates\flutter\debug/snapshot_blob.bin.d --asset-dir M:\lab\gilda\yabco_divar\build\app\intermediates\flutter\debug/flutter_assets --debug
```
## exception
```
FileSystemException: FileSystemException: Deletion failed, path = 'M:\lab\gilda\yabco_divar\build\app\intermediates\flutter\debug/flutter_assets' (OS Error: The directory is not empty.
, errno = 145)
```

```
#0      _Directory._deleteSync (dart:io/directory_impl.dart:206:7)
#1      FileSystemEntity.deleteSync (dart:io/file_system_entity.dart:466:47)
#2      ForwardingFileSystemEntity.deleteSync (package:file/src/forwarding/forwarding_file_system_entity.dart:72:16)
#3      writeBundle (package:flutter_tools/src/bundle.dart:228:15)
<asynchronous suspension>
#4      assemble (package:flutter_tools/src/bundle.dart:221:9)
<asynchronous suspension>
#5      build (package:flutter_tools/src/bundle.dart:154:9)
<asynchronous suspension>
#6      BuildBundleCommand.runCommand (package:flutter_tools/src/commands/build_bundle.dart:86:11)
<asynchronous suspension>
#7      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:527:18)
<asynchronous suspension>
#8      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:461:33)
<asynchronous suspension>
#9      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:142:29)
<asynchronous suspension>
#10     _rootRun (dart:async/zone.dart:1124:13)
#11     _CustomZone.run (dart:async/zone.dart:1021:19)
#12     _runZoned (dart:async/zone.dart:1516:10)
#13     runZoned (dart:async/zone.dart:1463:12)
#14     AppContext.run (package:flutter_tools/src/base/context.dart:141:18)
<asynchronous suspension>
#15     FlutterCommand.run (package:flutter_tools/src/runner/flutter_command.dart:452:20)
#16     CommandRunner.runCommand (package:args/command_runner.dart:196:27)
<asynchronous suspension>
#17     FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:401:21)
<asynchronous suspension>
#18     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:142:29)
<asynchronous suspension>
#19     _rootRun (dart:async/zone.dart:1124:13)
#20     _CustomZone.run (dart:async/zone.dart:1021:19)
#21     _runZoned (dart:async/zone.dart:1516:10)
#22     runZoned (dart:async/zone.dart:1463:12)
#23     AppContext.run (package:flutter_tools/src/base/context.dart:141:18)
<asynchronous suspension>
#24     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:361:19)
<asynchronous suspension>
#25     CommandRunner.run.<anonymous closure> (package:args/command_runner.dart:111:29)
#26     new Future.sync (dart:async/future.dart:224:31)
#27     CommandRunner.run (package:args/command_runner.dart:111:11)
#28     FlutterCommandRunner.run (package:flutter_tools/src/runner/flutter_command_runner.dart:241:18)
#29     run.<anonymous closure> (package:flutter_tools/runner.dart:60:20)
<asynchronous suspension>
#30     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:142:29)
<asynchronous suspension>
#31     _rootRun (dart:async/zone.dart:1124:13)
#32     _CustomZone.run (dart:async/zone.dart:1021:19)
#33     _runZoned (dart:async/zone.dart:1516:10)
#34     runZoned (dart:async/zone.dart:1463:12)
#35     AppContext.run (package:flutter_tools/src/base/context.dart:141:18)
<asynchronous suspension>
#36     runInContext (package:flutter_tools/src/context_runner.dart:44:24)
<asynchronous suspension>
#37     run (package:flutter_tools/runner.dart:51:10)
#38     main (package:flutter_tools/executable.dart:51:9)
<asynchronous suspension>
#39     main (file:///M:/flutter/packages/flutter_tools/bin/flutter_tools.dart:8:3)
#40     _startIsolate.<anonymous closure> (dart:isolate/runtime/libisolate_patch.dart:287:32)
#41     _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:171:12)
```

## flutter doctor

```
[✓] Flutter (Channel dev, v1.1.5, on Microsoft Windows [Version 6.3.9600], locale en-US)
    • Flutter version 1.1.5 at M:\flutter
    • Framework revision 63b2daff7f (9 days ago), 2018-12-28 10:43:06 -0800
    • Engine revision f0a1d6f91d
    • Dart version 2.1.1 (build 2.1.1-dev.0.1 d1817ddc91)

[✓] Android toolchain - develop for Android devices (Android SDK version 27.0.3)
    • Android SDK at D:\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-27, build-tools 27.0.3
    • ANDROID_HOME = D:\Android\sdk
    • Java binary at: C:\Program Files\Java\jdk1.8.0_45\bin\java
    • Java version Java(TM) SE Runtime Environment (build 1.8.0_45-b15)
    • All Android licenses accepted.

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).

[✓] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 6.0 (API 23) (emulator)

! Doctor found issues in 1 category.
```

**Comment by zoechi** [comment]
If the directory of file can't be deleted then there is probably some other app that has the file/directory open. Could be AntiVirus
Like for example https://github.com/flutter/flutter/issues/22316#issuecomment-426705145, https://github.com/flutter/flutter/issues/23295#issuecomment-432186004

**Comment by zoechi** [comment]
That AntiMalware was just an example. 
This issue is not very common and the error message is IMHO quite clear
and the cause is quite likely outside Flutter so printing advise is prone to be misleading.

**Comment by p30arena** [comment]
@zoechi 

This is the responsibility of the cross-platform software to handle platform specific errors too.

A proper message in this case, stops newbies/lazies from posting more issues, this is the only reason I've posted the issue, to prevent more issues.

**Comment by zoechi** [comment]
I think the problem is that the OS does not provide any information why it doesn't execute the delete command, so any other error message would just be guesswork.
I'll leave it to the specialists to decide how to proceed.

**Comment by christopherfujino** [comment]
Closed as a duplicate of #29867
